crates/driver/src/infra/simulator/mod.rs

```diff
@@ -152,6 +152,8 @@ pub enum SimulatorError {
     Blockchain(#[from] blockchain::Error),
     #[error("enso error: {0:?}")]
     Enso(#[from] enso::Error),
+    #[error("the simulated gas {0} exceeded the gas limit {1}")]
```

Comment:
nit: let's be slightly more explicit here to reduce the chance that somebody has to explain to the solver teams what this error means.
```suggestion
    #[error("the simulated gas {0} exceeded the gas limit {1} provided in the solution")]
```